spring-integration-core/src/main/java/org/springframework/integration/graph/MessageChannelInfo.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.graph;
+
+import org.springframework.integration.channel.AbstractMessageChannel;
+import org.springframework.integration.support.management.Statistics;
+import org.springframework.messaging.MessageChannel;
+
+/**
+ * Represents a message channel.
+ *
+ * @author Gary Russell
+ * @since 4.3
+ *
+ */
+public class MessageChannelInfo extends NodeInfo {
+
+	public MessageChannelInfo(MessageChannel channel) {
+		super(channel, channel instanceof AbstractMessageChannel ? new Stats((AbstractMessageChannel) channel) :
+			new NodeInfo.Stats());
+	}
+
+
+	public static class Stats extends NodeInfo.Stats {
+
+		private final AbstractMessageChannel channel;
+
+		private Stats(AbstractMessageChannel channel) {
+			this.channel = channel;
+		}
+
+		@Override
+		public boolean isAvailable() {
+			return this.channel.isCountsEnabled();
```

Comment:
Ah! Never mind. I see `isStatsEnabled()` below.
So, this one is just to indicate the whole state for the metrics. `setStatsEnabled()` enables the `counts`, too.
But `setCountsEnabled()` doesn't enable `stats`. They must be enable explicitly.